docs/build-as-dll.md

```diff
@@ -0,0 +1,8 @@
+# Design of building OpenTelemetry C++ SDK and exporters as DLL
+
+OpenTelemetry C++ can be built into single DLL named opentelemetry_cpp.dll on
+Windows, with CMake flag `OPENTELEMETRY_BUILD_DLL` set to `ON`.
+
+To instrument application based on the DLL, please define macro
+`OPENTELEMETRY_BUILD_IMPORT_DLL` before including any OpenTelemetry API header
+files in the application source files.
```

Comment:
See previous comment, defining _IMPORT_DLL can probably be avoided.

Reply:
As there are 3 status, I think define 2 internal variables could make it more clear:)